assignments/tasks.go

```diff
@@ -0,0 +1,284 @@
+// Updates an issue on specified repository
+func updateScmIssue(c context.Context, sc scm.SCM, course *pb.Course, repo *pb.Repository, issue *pb.Issue) (*scm.Issue, error) {
+	issueOptions := &scm.CreateIssueOptions{
+		Organization: course.Name,
+		Repository:   filepath.Base(repo.GetHTMLURL()), // Todo
+		Title:        issue.Title,
+		Body:         issue.Body,
+	}
+	return sc.EditRepoIssue(c, int(issue.IssueNumber), issueOptions)
+}
+
+// Creates an issue on specified repository.
+func createScmIssue(c context.Context, sc scm.SCM, course *pb.Course, repo *pb.Repository, task *pb.Task) (*scm.Issue, error) {
+	issueOptions := &scm.CreateIssueOptions{
+		Organization: course.Name,
+		Repository:   filepath.Base(repo.GetHTMLURL()), // Needs to be of type "tests", not "https://github.com/qf101/tests". This is a very hacky solution. pb.Repository should probably have a field "Name" that is set upon creation.
+		Title:        task.Title,
+		Body:         task.Body,
+	}
+	issue, err := sc.CreateIssue(c, issueOptions)
+	if err != nil {
+		return nil, err
+	}
+	return issue, nil
+}
+
+// This is more of a converter function, and cannot currently return an error. Should probably be renamed or something.
+func createIssue(c context.Context, repo *pb.Repository, task *pb.Task, scmIssue *scm.Issue) (*pb.Issue, error) {
+	issue := &pb.Issue{
+		RepositoryID: repo.ID,
+		IssueNumber:  uint64(scmIssue.IssueNumber),
+		Name:         task.Name,
+		Title:        task.Title,
+		Body:         task.Body,
+	}
+	return issue, nil
+}
+
+// Following is Oje code (placement might be temporary):
+
+func getTasksFromAssignments(c context.Context, assignments []*pb.Assignment) []*pb.Task {
+	tasks := []*pb.Task{}
+	for _, assignment := range assignments {
+		tasks = append(tasks, assignment.Tasks...)
+	}
+
+	return tasks
+}
+
+func handleTasks(c context.Context, db database.Database, s scm.SCM, course *pb.Course, tasks []*pb.Task) error {
+	repos, err := db.GetRepositoriesWithIssues(&pb.Repository{
+		OrganizationID: course.GetOrganizationID(),
+	})
+	if err != nil {
+		return err
+	}
+
+	assignments, err := db.GetAssignmentsByCourse(course.GetID(), false)
+	if err != nil {
+		return err
+	}
+
+	// Loops through all assignments
+	for _, assignment := range assignments {
+		err := synchronizeTasks(c, db, assignment, tasks)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Loops through all student repos
+	for _, repo := range repos {
+		if !repo.IsStudentRepo() {
+			continue
+		}
+
+		err = synchronizeIssues(c, db, course, s, repo, tasks)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// synchronizeTasks synchronizes tasks in the database, with the ones of given assignment.
+// Returns a slice of tasks as they appear in the database.
+func synchronizeTasks(c context.Context, db database.Database, assignment *pb.Assignment, tasks []*pb.Task) error {
+	tasksToBeCreated := []*pb.Task{}
+	tasksToBeUpdated := []*pb.Task{}
+	taskMap := make(map[string]*pb.Task)
```

Comment:
Could instead pass in the `taskMap` directly instead of `[]*pb.Tasks`... it is not totally clear to me if you need the check for `assignment.Order`? If needed, you can still iterate over the map I guess.